### PR TITLE
guard against divide by 0 when splitting parallelism

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
We saw a spike in divide by zero panics in the code introduced in #7769. I was able to reproduce this error via a test that calculates `WeightedParallelism` with a start that's after the end. Not sure if this is possible, but we definitely saw this happening in our ops environment, so something is causing it, and the fix should guard against it in any case.

**Special notes for your reviewer**:

**Checklist**
- [X] Tests updated

### CODE REVIEW

**Review by MasslessParticle — APPROVED** [review]

**Review by dannykopping — COMMENTED** [review]
It seems to me like we should be returning an error to indicate that an expectation failed (such as end time before start time) and let the calling function handle this gracefully with an error.

Returning 0 or 1 here doesn't carry a lot of semantic meaning upon first glance, but admittedly I'm not too familiar with this code.

**Comment by sandeepsukhani** [comment]
> It seems to me like we should be returning an error to indicate that an expectation failed (such as end time before start time) and let the calling function handle this gracefully with an error.
> 
> Returning 0 or 1 here doesn't carry a lot of semantic meaning upon first glance, but admittedly I'm not too familiar with this code.

The reason we are not handling all the negative cases is to avoid having checks all over the place. This code is supposed just to find the right parallelism parameter based on its overlap across tsdb vs non-tsdb schema configs. If we do all the checks here, then we will have to ensure the same checks happen both here and in querier.

File: pkg/querier/queryrange/limits.go [hunk 1]
```diff
@@ -366,11 +366,13 @@ func WeightedParallelism(
 	tsdbMaxQueryParallelism := l.TSDBMaxQueryParallelism(user)
 	regMaxQueryParallelism := l.MaxQueryParallelism(user)
 	if tsdbMaxQueryParallelism+regMaxQueryParallelism == 0 {
+		level.Info(util_log.Logger).Log("msg", "querying disabled for tenant")
 		return 0
 	}
 
 	// query end before start would anyways error out so just short circuit and return 1
 	if end < start {
+		level.Warn(util_log.Logger).Log("msg", "query end time before start, letting downstream code handle it gracefully", "start", start, "end", end)
```
Comment (dannykopping) [inline pkg/querier/queryrange/limits.go:375]: Do we include the tenant in these logs? And how will we be able to identify which query caused this warning log?

**Review by dannykopping — COMMENTED** [review]
LGTM but I'm struggling to see how these logs will be operationally useful in this current form

File: pkg/querier/queryrange/limits.go (hunk 1 above)
Comment (sandeepsukhani) [reply pkg/querier/queryrange/limits.go]: yeah, good callout! I have made some changes to log `orgID` and `traceID` with the logs.

**Review by dannykopping — APPROVED** [review]
LGTM!

**Comment by trevorwhitney** [comment]
Great work @sandeepsukhani, thanks for picking this up and making it better!

**Pull request merged**